exonum-light-client/src/main/java/com/exonum/client/response/BlocksResponse.java

```diff
@@ -28,14 +28,17 @@
   List<Block> blocks;
 
   /**
-   * The smallest inclusive height of the returned blocks that match the search criteria.
+   * The smallest height of the returned blocks that match the search criteria.
    */
   long blocksRangeStart;
 
   /**
-   * The largest exclusive height of the returned blocks that match the search criteria.
+   * The largest height of the returned blocks that match the search criteria.
+   * The value is equal to {@code heightMax + 1} if heightMax request parameter is passed.
```

Comment:
If I pass heightMax, but at that height the block is empty and filtered do I always get `blocks[0].height + 1` or do I get `heightMax + 1`?

Reply:
I might have not been clear in the question, but what would happen if

```
Blockchain: 0 - 1 - 2 - 3 - 4 - 5 - <and goes on> - N
                    e       e

Request:  latest=4, count=2, skip_empty

In blocks I would get [B3, B1].

But what do I get for start/end:
 - start=1, end=5 (latest + 1, from request)
 - start=1, end=4 (last matching + 1)
```

---

What if we remove it completely till 0.12? Or, if it is required internally, add a <b>Warning: the value is set mostly randomly — read below if would still like to use.</b>

Reply:
in this case, you will get `blocks[0].heigth + 1` i.e. `end=4` (I believe ).